### PR TITLE
Add Dell Enterprise SONiC 'pim_interfaces' module

### PR DESCRIPTION
**Pull request by ArunSaravananBalachandran**
##### SUMMARY

- Add Enterprise SONiC 'pim_interfaces' resource module:
  - Provide data definitions in 'argspec'.
  - Implement 'facts' to gather facts for the module.
  - Implement 'config' to apply configurations for merged, deleted, replaced and overridden states.
  - Implement support for 'check' and 'diff' modes.
- Define regression test cases to verify correct functionality of pim_interfaces resource module.

##### GitHub Issues
List the GitHub issues impacted by this PR. If no Github issues are affected, please indicate this with "N/A".

| GitHub Issue # |
| -------------- |
| N/A |

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

- sonic_pim_intefaces

##### Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [ ] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

##### How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Regression report: [pim_interfaces_regression_report.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/15097948/pim_interfaces_regression_report.pdf)

### CODE REVIEW

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
Thank you for providing this implementation of the "PIM interfaces" resource module.

The code structure and test results look good.

I have no questions or comments on the instructions in the proposed change set. I have posted only minor documentation change requests.

File: plugins/modules/sonic_pim_interfaces.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+---
+module: sonic_pim_interfaces
+version_added: '3.0.0'
+notes:
+  - Supports C(check_mode).
+short_description: Manage interface-specific PIM configurations on SONiC
+description:
+  - This module provides configuration management of interface-specific
+    PIM parameters for devices running SONiC.
+  - BFD profile need to be created earlier in the device.
```
[the remark this answers is not in the document]
Comment (ArunSaravananBalachandran) [reply plugins/modules/sonic_pim_interfaces.py]: Done.

File: plugins/modules/sonic_pim_interfaces.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+#  ip pim bfd
+#  ip pim bfd profile profile_1
+# sonic# show running-configuration interface Eth 1/2 | grep "ip pim"
+# sonic#
+
+
+"""
+RETURN = """
+before:
+  description: The configuration prior to the model invocation.
```
[the remark this answers is not in the document]
Comment (ArunSaravananBalachandran) [reply plugins/modules/sonic_pim_interfaces.py]: Done.

File: plugins/modules/sonic_pim_interfaces.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+  returned: always
+  type: list
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
```
[the remark this answers is not in the document]
Comment (ArunSaravananBalachandran) [reply plugins/modules/sonic_pim_interfaces.py]: Done.

File: plugins/modules/sonic_pim_interfaces.py [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+after:
+  description: The resulting configuration model invocation.
+  returned: when changed
+  type: list
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
```
[the remark this answers is not in the document]
Comment (ArunSaravananBalachandran) [reply plugins/modules/sonic_pim_interfaces.py]: Done.

File: plugins/modules/sonic_pim_interfaces.py [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+after(generated):
+  description: The generated configuration model invocation.
```
[the remark this answers is not in the document]
Comment (ArunSaravananBalachandran) [reply plugins/modules/sonic_pim_interfaces.py]: Done.

File: plugins/modules/sonic_pim_interfaces.py [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+  returned: when C(check_mode)
+  type: list
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
```
[the remark this answers is not in the document]
Comment (ArunSaravananBalachandran) [reply plugins/modules/sonic_pim_interfaces.py]: Done.

**Comment by kerry-meyer** [comment]
Thanks for making the incremental changes for this change set.

The revised code looks good, and the posted test results look good.

Thank you for implementing the "PIM Interfaces" resource module.

To proceed with approval and merging of this PR, please resolve the merge conflicts and notify me. I will Approve and merge the PR after that.

**Review by kerry-meyer — APPROVED** [review]
Thanks for making the incremental changes for this change set and re-synching the PR branch to merge in recent changes in the common files.

The revised code looks good, and the posted test results look good.

Thank you for implementing the "PIM Interfaces" resource module.

**Pull request merged**